platform/ios/docs/guides/Data-Driven Styling.md

```diff
@@ -0,0 +1,127 @@
+* `MGLSourceStyleFunction` - For a style value that changes with the attributes of a feature. You can adjust the radius of a circle based on the magnitude of an earthquake, for example.
+* `MGLCompositeStyleFunction` - For a style value that changes with both zoom level and attribute values. For example, you can add a circle layer where each circle has a radius based on both zoom level and the magnitude of an earthquake.
+
+####Stops
+
+Stops are key-value pairs that that determine a style value. With a `MGLCameraSourceFunction` stop, you can use a dictionary with a zoom level for a key and a `MGLStyleValue` for the value. For example, you can use a stops dictionary with zoom levels 0, 10, and 20 as keys, and yellow, orange, and red as the values. A `MGLSourceStyleFunction` uses the relevant attribute value as the key.
+
+####`MGLInterpolationMode`
+
+The effect a key has on the style value is determined by the interpolation mode. There are four interpolation modes that can be used with a source style function: exponential, interval, categorical, and identity. You can also use exponential and interval interpolation modes with a camera style function.
+
+#####Exponential:
+
+`MGLInterpolationModelExponential` creates a linear effect based on the values. The key value is the base for interpolation, and the style value is based on where an attribute value falls between two keys.
+
+The stops dictionary below, for example, shows colors that continuously shift from yellow -> orange -> red -> blue -> white based on the attribute value.
+
+```
+  if let url = URL(string: "https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_week.geojson") {
+            let symbolSource = MGLSource(identifier: "source")
+            let symbolLayer = MGLSymbolStyleLayer(identifier: "place-city-sm", source: symbolSource)
+
+            let source = MGLShapeSource(identifier: "earthquakes", url: url, options: nil)
+            style.addSource(source)
+
+            let stops = [0 : MGLStyleValue(rawValue: UIColor.yellow),
+                         2.5 : MGLStyleValue(rawValue: UIColor.orange),
+                         5: MGLStyleValue(rawValue: UIColor.red),
+                         7.5 : MGLStyleValue(rawValue: UIColor.blue),
+                         10 : MGLStyleValue(rawValue: UIColor.white)]
+
+            let layer = MGLCircleStyleLayer(identifier: "circles", source: source)
+            layer.circleColor = MGLStyleValue<UIColor>(interpolationMode: .exponential,
+                                                          sourceStops: stops,
+                                                          attributeName: "mag",
+                                                          options: [.defaultValue: MGLStyleValue<UIColor>(rawValue: .green)])
```

Comment:
Oh, good point. I will see if I can adjust the example to show that. Alternatively, we can use it in an upcoming example